File: app/src/main/java/com/mbmc/fiinfo/ui/fragment/CodeInstructionsFragment.java
package com.mbmc.fiinfo.ui.fragment;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.v7.app.AlertDialog;
import android.view.View;
import android.widget.CheckBox;

import com.mbmc.fiinfo.R;
import com.mbmc.fiinfo.constant.Preferences;
import com.mbmc.fiinfo.helper.CodeManager;
import com.mbmc.fiinfo.helper.PreferencesManager;


public class CodeInstructionsFragment extends DialogFragment {

    private int code;


    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        View view = View.inflate(getActivity(), R.layout.view_code_instructions, null);
        final CheckBox checkBox = (CheckBox) view.findViewById(R.id.code_instructions_check_box);

        return new AlertDialog.Builder(getActivity(), R.style.AlertDialogTheme)
                .setView(view)
                .setPositiveButton(R.string.ok, (DialogInterface dialogInterface, int which) -> {
                    if (checkBox.isChecked()) {
                        PreferencesManager.getInstance(getActivity())
                                .setBoolean(Preferences.HIDE_CODE_INSTRUCTIONS, true);
                    }
                    CodeManager.openDialer(getActivity(), code);
                })
                .setNegativeButton(R.string.cancel, (DialogInterface dialogInterface, int which) -> dialogInterface.dismiss())
                .create();
    }

    public void setCode(int code) {
        this.code = code;
    }

}

File: app/src/main/java/com/mbmc/fiinfo/ui/codes/CodeInstructionsFragment.kt
package com.mbmc.fiinfo.ui.codes

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import com.mbmc.fiinfo.R
import com.mbmc.fiinfo.data.Code
import com.mbmc.fiinfo.databinding.FragmentCodeInstructionsBinding
import com.mbmc.fiinfo.helper.CodeManager
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class CodeInstructionsFragment(private val code: Code) : DialogFragment() {

    private lateinit var binding: FragmentCodeInstructionsBinding
    @Inject lateinit var codeManager: CodeManager

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        binding = FragmentCodeInstructionsBinding.inflate(layoutInflater)
        return AlertDialog.Builder(requireContext())
            .setView(binding.root)
            .setNegativeButton(R.string.cancel) { _, _ -> }
            .setPositiveButton(R.string.ok) { _, _ ->
                if (binding.hide.isChecked) {
                    codeManager.setHasSeenInstructions()
                }
                codeManager.openDialer(requireContext(), code.code)
            }
            .create()
    }

    companion object {
        const val TAG = "CodeInstructionsFragment"
    }
}